spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -327,7 +327,11 @@ public void pruneColumns(StructType requestedSchema) {
     StructType requestedProjection =
         new StructType(
             Stream.of(requestedSchema.fields())
-                .filter(field -> MetadataColumns.nonMetadataColumn(field.name()))
+                .filter(
+                    field ->
+                        MetadataColumns.nonMetadataColumn(field.name())
+                            || field.name().equals("_row_id")
```

Comment:
would be good to use `MetadataColumns.ROW_ID.name()` here

Reply:
Actually this code path is no longer needed, removed it entirely!